### ISSUE TITLE
slp1 to slp1_accented in transliteration-- Side effects!!

### ISSUE THREAD

**Issue by Andhrabharati** [sanskrit-lexicon/csl-devanagari#38]
@drdhaval2785 

Just noticed that this new transliteration code has unwanted effects mainly in BOR & SKD, where no accent is involved, but just a slash in normal sense is intended.

**Comment by Andhrabharati** [comment]
@drdhaval2785 

Even VCP is heavily affected by this change.

**Comment by drdhaval2785** [comment]
@vvasuki 
Do you have any solution which will not produce undesirable side effects of accents?

**Comment by vvasuki** [comment]
> @vvasuki Do you have any solution which will not produce undesirable side effects of accents?

Yes - Don't use slp1_accented on dictionaries which don't have accents! Garbage in - garbage out.

Going back to a more basic design issue - why do you keep SLP1 encoding in the dicts in the first place? Maybe back in the day, unicode devanAgarI standard was not popular, so they had to do such monkey tricks. But in 2022, one can save devanAgarI data directly using devanAgarI unicode.

**Comment by drdhaval2785** [comment]
Can you specify which dictionaries have accents and which do not? I will make modifications accordingly.

It is not at all feasible to keep data in Devanagari unicode without unnecessary hassles. So SLP1 is going to stay for long time. I really look forward to a day when Devanagari Unicode would emulate Sanskrit consonants and vowels more naturally. It

**Comment by vvasuki** [comment]
> Can you specify which dictionaries have accents and which do not? I will make modifications accordingly.

शब्दकल्पद्रुमः, वाचस्पत्यं च। अन्येऽपि स्युर् बहवः - ये जानन्ति, ते वदेयुः। चिता एव कोशाः स्वरं दर्शयन्ति। 

> 
> It is not at all feasible to keep data in Devanagari unicode without unnecessary hassles. So SLP1 is going to stay for long time. I really look forward to a day when Devanagari Unicode would emulate Sanskrit consonants and vowels more naturally. It

Sentence is broken in the middle?

Anyway, use SLP1 or ISCII or ... for internal processing as needed however much you like. You don't need to store textual data in it - that's what leads to avoidable problems such as this.

EDIT: If you digitized SKD or VSP, you would use devanAgarI unicode! (as you know from your kosha project)

**Comment by Andhrabharati** [comment]
@drdhaval2785 

Incidentally, even the tags and English text in those places (within the body matter) got converted to Devanagari, in those dictionaries.

This point also needs to be addressed.

**Comment by drdhaval2785** [comment]
I would appreciate examples

**Comment by Andhrabharati** [comment]
In VCP,
`<फ्>	for <P>`
`<ःई>	for <HI>`
`<एदित् त्य्पे="ह्ट्"꣡>	for <edit type="hw"/>`
`<छ्[०-९]+	for <C[0-9]+`
`<फिच्तुरे>	for <Picture>`

In SKD,
`<ॠ>	for  <F>`
`<꣡ॠ>	for  </F>`
`<फिच्तुरे>	for <Picture>`

**Also SKD (in contrast) has quite a few `<H>` lines remained in slp1, unconverted to Devanagari.**

And interestingly KRM has no such tag conversion issue.

As these are not related to the accent mark, guess they need spl. attention even with slp1 conversion!

**Comment by Andhrabharati** [comment]
Surprisingly, even the MW has fell a victim of this "tag conversion"!
`<स्र्स्꣡> for <srs1>`
`<स्होर्त्लोन्ग्꣡> for <shortlong1>`

**Comment by vvasuki** [comment]
These tag issues could be because @drdhaval2785 's scripts are not passing some toggler arguments (which are no longer set by default) - https://github.com/indic-transliteration/indic_transliteration_py/blob/1ba2688d235eccc0c5ac629c46ac9df83ef331f7/indic_transliteration/sanscript/__init__.py#L189 . Also, suitable togglers can be used to leave non-svara-encoding `/` marks alone.

**Comment by Andhrabharati** [comment]
yes, I understand it.

I was informing him these tags, to be marked suitably similar to many other tags that are out of the purview of transliteration.

**Comment by drdhaval2785** [comment]
I am not aware when indic_transliteration package started to require explicit togglers. I never had similar problem earier. Maybe some version update introduced this artefact.

Will correct soon.

**Comment by Andhrabharati** [comment]
@drdhaval2785 

just fyi (if you didn't notice it earlier)--

this indic_transliteration package can generate iast output as well, in addition to various other scripts (apart fron Devanagari).

**Comment by gasyoun** [comment]
<details>
<summary>🤖 Bug triage findings</summary>

**Bug class:** display-rendering (transcoding side-effect — XML markup corruption in generated output files)

**Evidence:**
The bug is confirmed present in the current generated output files in v02/:

- v02/vcp/vcp.txt: **2,321 occurrences** of <फ्> (should be <P>) — e.g. line 3900
- v02/skd/skd.txt: **5 occurrences** of <ॠ> (should be <F>) — e.g. line 79028
- v02/mw/mw.txt: **32,981 occurrences** of <स्र्स्꣡> (should be <srs1>) and similar corrupted attribute tags

Root cause confirmed in scripts/to_devanagari.py lines 55-71 (convert_to_devanagari): for VCP, SKD, and armh the entire body line is passed through sanscript.transliterate(lin, 'slp1_accented', 'devanagari') with no protection for XML tags. For MW (convert_partially_to_devanagari) the regex split on <s>...</s> only protects Sanskrit spans; tag attributes outside those spans (srs1, shortlong1, edit type=hw) are transliterated.

The slp1_accented scheme also maps / to Vedic accent character, corrupting plain-slash uses in BOR and other non-accented dicts.

**Agent-fixable (Tier A)?** Partially — the script fix is clear; regenerating large output files (VCP ~136k lines, SKD ~200k lines, MW ~630k lines) requires running the pipeline.

**Proposed fix:**

In scripts/to_devanagari.py, replace 'slp1_accented' with 'slp1' for dicts that do not carry Vedic accent marks. The accented scheme is only needed for GRA and ACC. Additionally, use the toggler/ignore arguments from indic_transliteration (as @vvasuki suggested referencing sanscript/__init__.py line 189) to protect XML tag content from transliteration.

Quick fix for convert_to_devanagari (line 69):
```python
ACCENTED_DICTS = ['acc', 'gra']
scheme = 'slp1_accented' if dictcode in ACCENTED_DICTS else 'slp1'
result.append(sanscript.transliterate(lin, scheme, 'devanagari'))
```

Apply symmetrically in to_slp1.py. After fixing, regenerate v02/ output for vcp, skd, armh, mw, bor, and other affected dicts.

**Suggested next step:** Fix slp1_accented -> slp1 in scripts/to_devanagari.py and to_slp1.py for non-accented dicts, then regenerate affected outputs.

</details>

**Comment by gasyoun** [comment]
Opened PR https://github.com/sanskrit-lexicon/csl-devanagari/pull/45 for this.

Summary: `to_devanagari.py` now preserves XML-like tags while transliterating SLP1-accented text nodes, with focused unit coverage for plain text, `<s>...</s>` segments, embedded XML tags, whole-line conversion, and skipped metadata lines.

Validation: `python -m unittest discover -s scripts -p "test*.py"` = 5 passed.

Note: this is the converter-script fix only; regenerating generated `v02/*` outputs remains server-side/follow-up work after the script patch lands.
